### PR TITLE
Transition CRDs from v1beta1 to v1

### LINKED ISSUE #1296: Transition from apiextensions.k8s.io/v1beta1 to apiextensions.k8s.io/v1
**What happened**:
When installing the Kubefed CRDs into a cluster, the API server will issue a warning:
```
Warning: apiextensions.k8s.io/v1beta1 CustomResourceDefinition is deprecated in v1.16+, unavailable in v1.22+; use apiextensions.k8s.io/v1 CustomResourceDefinition
```

**What you expected to happen**:
No warnings

**How to reproduce it (as minimally and precisely as possible)**:
```
kubectl create -f charts/kubefed/crds/crds.yaml
```

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`): 1.19.1
- KubeFed version: `master`
- Scope of installation (namespaced or cluster)
- Others

/kind bug

### PR DESCRIPTION
**Pull request by makkes**
**What this PR does / why we need it**:
Transitions all the CRDs defined in the kubefed chart and sub-charts as well as those generated through `kubefedctl` from apiextensions.k8s.io/v1beta1 to v1.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #1296

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by makkes** [comment]
@jimmidyson @hectorj2f @xunpan the CI checks are passing now (finally 🎉). Would you mind having a look at this PR, please?

File: pkg/kubefedctl/enable/validation.go [hunk 1]
```diff
@@ -115,27 +116,7 @@ func federatedTypeValidationSchema(templateSchema map[string]v1beta1.JSONSchemaP
...
-														Type: "boolean",
-													},
-													{
-														Type: "object",
-													},
-													{
-														Type: "array",
-													},
-												},
+												XPreserveUnknownFields: pointer.BoolPtr(true),
```
Comment (jimmidyson) [inline pkg/kubefedctl/enable/validation.go:119]: Do we need:
```suggestion
												XPreserveUnknownFields: pointer.BoolPtr(true),
												Type:                   "object",
```

**Review by jimmidyson — COMMENTED** [review]
One question, but otherwise LGTM. Wonder if there is a nice way to also upgrade already deployed federated* CRDs to v1 in the same way when upgrading kubefed? Do we even need to worry about that?

**Comment by makkes** [comment]
Good point @jimmidyson. I will have a look into the upgrade behaviour. Do you know off the top of your head whether there's any upgrade testing in the e2e tests already?

**Comment by hectorj2f** [comment]
@makkes I can confirm, there is not any upgrade testing.

File: pkg/kubefedctl/enable/validation.go (hunk 1 above)
Comment (makkes) [reply pkg/kubefedctl/enable/validation.go:119]: Honestly, I don't know. It works without it, too. 😬

**Comment by makkes** [comment]
@jimmidyson: I was finally able to test the upgrade from a former version of kubefed:

* checked out current `master`
* ran `make deploy.kind`
* attached another kind cluster
* created a `Deployment` and federated it using `kubefedctl`
* checked out this PR's branch and ran `make deploy.kind` again, updating the CRDs and controllers/webhooks
* created a 2nd `Deployment` and federated it using the new `kubefedctl` binary

No interruption of the federated resources was observed and all new resources are federated properly.

**Comment by jimmidyson** [comment]
Thanks @makkes for testing so thoroughly.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,8 @@
 # Unreleased
 
+- fix: Transition from apiextensions.k8s.io/v1beta1 to
```
Comment (jimmidyson) [inline CHANGELOG.md:3]: I'd say this was a feature rather than a fix btw.

Comment (makkes) [reply CHANGELOG.md]: Fixed.

**Review by hectorj2f — APPROVED** [review]

**Pull request merged**